lib/speech_to_text/ibm.rb

```diff
@@ -4,12 +4,39 @@
 # BigBlueButton open source conferencing system - http://www.bigbluebutton.org/
 #
 # Copyright (c) 2019 BigBlueButton Inc. and by respective authors (see below).
-#
+
+require "speech_to_text"
+require("ibm_watson/speech_to_text_v1")
 require_relative "util.rb"
 
 module SpeechToText
 	module IbmWatsonS2T
 		include Util
+
+		#create new job on watson server by uploading audio
+		#function returns 2 variables IBMWatson::SpeechToTextV1 object and jobid
+	  def self.create_job(published_files,recordID,apikey)
+	    speech_to_text = IBMWatson::SpeechToTextV1.new(iam_apikey: apikey)
+	    audio_file = File.open("#{published_files}/#{recordID}/#{recordID}.flac")
+	    service_response = speech_to_text.create_job(audio: audio_file,content_type: "audio/flac", timestamps: true)
+	    job_id = service_response.result["id"]
+	    return speech_to_text,job_id
+	  end
+
+		#functions checks the status of specific jobid
+		#pass array of 2 variables as argumanet [IBMWatson::SpeechToTextV1 object, jobid]
+	  def self.check_job(params)
+	    status = "processing"
+			speech_to_text = params[0]
+			job_id = params[1]
+	    while(status != "completed")
+	      service_response = speech_to_text.check_job(id: job_id)
+	      status = service_response.result["status"]
```

Comment:
Don't loop here. Just return {status, job_id} if not status != completed. If completed, return {status, job_id, result}